src/service/document-info-impl.js

```diff
@@ -95,3 +102,33 @@ export class DocInfo {
 function getPageViewId(win) {
   return String(Math.floor(win.Math.random() * 10000));
 }
+
+/**
+ * Returns a map object of link tag relations in document head.
+ * Key is the link rel, value is a list of corresponding hrefs.
+ * @param {!Document} doc
+ * @return {!Object<string, !Array<string>>}
+ */
+function getLinkRels(doc) {
+  const linkRels = map();
+  if (doc.head) {
+    const links = doc.head.querySelectorAll('link');
+    for (let i = 0; i < links.length; i++) {
+      const link = links[i];
+      const rel = link.getAttribute('rel');
```

Comment:
need to handle the case where the rel has multiple mappings to the same href.

`<link rel="sharelink canonical">`

`split(/\s+/)`